src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs

```diff
@@ -1,66 +1,83 @@
-﻿using Microsoft.AspNet.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Composition.Hosting;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
 using OmniSharp.Dnx;
 using OmniSharp.Models;
 using OmniSharp.Models.v1;
-using OmniSharp.MSBuild;
-using OmniSharp.ScriptCs;
+using OmniSharp.Services;
 
 namespace OmniSharp
 {
     public class ProjectSystemController
     {
-        private readonly DnxContext _dnxContext;
         private readonly OmnisharpWorkspace _workspace;
-        private readonly MSBuildContext _msbuildContext;
-        private readonly ScriptCsContext _scriptCsContext;
+        private readonly IEnumerable<IProjectSystem> _projectSystems;
 
-        public ProjectSystemController(DnxContext dnxContext, MSBuildContext msbuildContext, ScriptCsContext scriptCsContext,
-            OmnisharpWorkspace workspace)
+        public ProjectSystemController(OmnisharpWorkspace workspace, CompositionHost host)
         {
-            _dnxContext = dnxContext;
-            _msbuildContext = msbuildContext;
-            _scriptCsContext = scriptCsContext;
+            _projectSystems = host.GetExports<IProjectSystem>();
             _workspace = workspace;
         }
 
         [HttpPost("/projects")]
         [HttpGet("/projects")]
-        public WorkspaceInformationResponse ProjectInformation(ProjectInformationRequest request)
+        public async Task<WorkspaceInformationResponse> ProjectInformation(ProjectInformationRequest request)
         {
-            return new WorkspaceInformationResponse
+            var tasks = new List<Task<Action>>();
+            var response = new WorkspaceInformationResponse();
+
+            foreach (var projectSystem in _projectSystems)
+            {
+                tasks.Add(projectSystem.GetInformationModel(request)
```

Comment:
I was concerned that there could be a threading issue, adding the items into the dictionary.